presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunner.java

```diff
@@ -0,0 +1,384 @@
+                char ch = line.charAt(i);
+                if (ch == '"') {
+                    // Handle double quotes inside quoted string
+                    if (insideQuotes && i + 1 < line.length() && line.charAt(i + 1) == '"') {
+                        currentField.append(ch);
+                        i++;
+                    }
+                    else {
+                        insideQuotes = !insideQuotes;
+                    }
+                }
+                else if (ch == ',' && !insideQuotes) {
+                    currentRow.add(currentField.toString());
+                    currentField.setLength(0); // Clear the current field
+                }
+                else {
+                    currentField.append(ch);
+                }
+            }
+            if (insideQuotes) {
+                currentField.append('\n'); // Add newline for multiline fields
+            }
+            else {
+                currentRow.add(currentField.toString());
+                currentField.setLength(0); // Clear the current field
+                allRows.add(currentRow.toArray(new String[0]));
+                currentRow.clear();
+            }
+        }
+        if (!currentRow.isEmpty()) {
+            currentRow.add(currentField.toString());
+            allRows.add(currentRow.toArray(new String[0]));
+        }
+        return allRows;
+    }
+
+    private static Type inferType(List<String> values)
+    {
+        boolean isBigint = true;
+        boolean isDouble = true;
+        boolean isBoolean = true;
+
+        for (String value : values) {
+            if (!value.matches("^-?\\d+$")) {
+                isBigint = false;
+            }
+            if (!value.matches("^-?\\d+\\.\\d+$")) {
+                isDouble = false;
+            }
+            if (!value.equalsIgnoreCase("true") && !value.equalsIgnoreCase("false")) {
+                isBoolean = false;
+            }
+        }
+
+        if (isBigint) {
+            return BigintType.BIGINT;
+        }
+        else if (isDouble) {
+            return DoubleType.DOUBLE;
+        }
+        else if (isBoolean) {
+            return BooleanType.BOOLEAN;
+        }
+        else {
+            return VarcharType.VARCHAR;
+        }
+    }
+
+    private static Object convertToType(String value, Type type)
+    {
+        if (value.isEmpty()) {
+            return null;
+        }
+        if (type.equals(VarcharType.VARCHAR)) {
+            return value;
+        }
+        else if (type.equals(BigintType.BIGINT)) {
+            return Long.parseLong(value);
+        }
+        else if (type.equals(DoubleType.DOUBLE)) {
+            return Double.parseDouble(value);
+        }
+        else if (type.equals(BooleanType.BOOLEAN)) {
+            return Boolean.parseBoolean(value);
+        }
+        else {
+            throw new IllegalArgumentException("Unsupported type: " + type);
+        }
+    }
+
+    public Container.ExecResult executeQuery(String sql)
```

Comment:
Can you please inject the default catalog and schema into this class so it's not hardcoded?

Reply:
Changed catalog and schema as system parameters.